src/Chapter2.hs

```diff
@@ -605,7 +632,8 @@ Implement a function that duplicates each element of the list
 
 -}
 duplicate :: [a] -> [a]
-duplicate = error "duplicate: Not implemented!"
+duplicate [] = []
+duplicate (x:xs) = [x, x] ++ duplicate xs
```

Comment:
Usually, adding elements to the end of the list is costly. Your implementation is correct :+1: But you can try to implement this function more efficiently by using `:` instead `++`. 

It is more efficient to construct lists using the cons operator directly without introducing intermediate values.
```suggestion
duplicate (x:xs) = x : x : duplicate xs
```